### PR TITLE
BENCH: adjust BenchGlobal

### PR DESCRIPTION
**Pull request by andyfaff**
Make it simpler to run `optimize.BenchGlobal`, and provide processing functionality to convert the results into pandas DataFrame

### CODE REVIEW

File: benchmarks/benchmarks/optimize.py [hunk 1]
```diff
@@ -506,7 +507,7 @@ class BenchGlobal(Benchmark):
         _enabled_functions = list(_functions.keys())
 
     params = [
-        list(_functions.keys()),
+        _enabled_functions,
```
Comment (andyfaff) [inline benchmarks/benchmarks/optimize.py:510]: There's no need to add all the functions to the list of params, only send those you want to run.

**Comment by dschmitz89** [comment]
Will try to run this locally in the next days.

**Comment by mdhaber** [comment]
@andyfaff I was using `BenchGlobal` tonight for gh-21314 and one thing I didn't understand was how the random state is controlled in `initial_vector` (which light only be used by `basinhopping`, but that's all I was looking at). It looks like the seed is set in `benchmarks` `__init.py__`, but then won't the actual state depend on execution order? 

Another suggestion would be to use `time.perf_counter` instead of `time.time` if ASVs timer can't be used. (I think I was able to use it and track objective function value efficiently in the `linprog` benchmarks, but I know it's tricky and the stochastic solvers have additional difficulties.) `time.time` has very low resolution on Windows. 

Any recommendations for running these problems manually, without ASV?

**Comment by andyfaff** [comment]
This is an under-visited part of the codebase.

> one thing I didn't understand was how the random state is controlled in initial_vector

`basinhopping` is the only global optimiser that uses `x0` (by default). It's therefore the only minimiser that needs randomisation of initial vectors. `initial_vector` uses the global `RandomState` rather than passing in a generator. Global RandomState is set via `__init__.py`. I guess the random seeding is to ensure uniform distribution of starting points rather than reproducibility.

I don't know how to run the benchmarks without having asv installed, not sure if there's machinery to do that. I run the benchmarks via the extra comment I added in this PR. I think it uses a build of the current state of the repo.

**Pull request merged**